File: scraper.js
const axios = require('axios');
const fs = require('fs');
const moment = require('moment');

const board = 'TQBR';
const yesterday = moment().subtract(1, 'days').format("YYYY-MM-DD");

const TICKs = fs.readFileSync("C:/Dor/courwo/TICK.txt", "utf8").split('\n').map(line => line.trim());

const outputFilePath = 'C:/Dor/courwo/Database//tickers_data.txt';
const output_file = fs.createWriteStream(outputFilePath);

async function fetchData() {
    for (const TICK of TICKs) {
            const response = await axios.get('http://iss.moex.com/iss/securities//');
            const data = response.data;
            const tickerData = data.map(row => `,${row.CLOSE}\n`).join('');
            output_file.write(tickerData);
    }
    output_file.end();
}
fetchData();

File: scraper.py
import requests
import datetime
import pathlib
import apimoex
import pandas as pd

board = 'TQBR'
current_time = datetime.datetime.now().time()

yesterday = datetime.date.today() - datetime.timedelta(days=1)

with open("C:/Dor/courwo/TICK.txt", "r") as TICKs:
    TICKs = [line.rstrip() for line in TICKs]

output_file = open("C:/Dor/courwo/Database/{}/tickers_data.txt".format(board), "w")
process = 0

with requests.session() as session:
    for TICK in TICKs:
        process += 1
        print((process / len(TICKs)) * 100, ' %')
        
        data = apimoex.get_board_history(session, TICK, board=board, start=yesterday, end=yesterday)
        
        if data:
            df = pd.DataFrame(data)
            df = df[['TRADEDATE', 'CLOSE']]
            for index, row in df.iterrows():
                ticker_data = "{}, {}\n".format(TICK, row['CLOSE']) 
                output_file.write(ticker_data)
        else:
            print(f"No data for {TICK}")

output_file.close()
